mission_control/navigator_missions/nav_missions/start_gate.py

```diff
@@ -101,45 +92,60 @@ def main(navigator):
 
     angle_buoys = np.array([np.arctan2(buoy[1], buoy[0]) for buoy in b_bl_buoys])
     b_left_buoy, b_right_buoy = back[np.argmax(angle_buoys)], back[np.argmin(angle_buoys)]
- 
-    points = [navigator_tools.numpy_to_point(b.position) for b in [f_left_buoy]]
+
 
     # Lets plot a course, yarrr
     f_between_vector, f_direction_vector = get_path(f_left_buoy, f_right_buoy)
     f_mid_point = f_left_buoy.position + f_between_vector / 2
+
     b_between_vector, _ = get_path(b_left_buoy, b_right_buoy)
     b_mid_point = b_left_buoy.position + b_between_vector / 2
-    through_vector = b_mid_point - f_mid_point
-    through_vector = through_vector / np.linalg.norm(through_vector)
+    
+    setup = navigator.move.set_position(f_mid_point).look_at(b_mid_point).backward(setup_dist)
+    target = setup.set_position(b_mid_point).forward(setup_dist)
+    
+    ogrid = OgridFactory(f_left_buoy.position, f_right_buoy.position, target.position,
+                         left_b_pos=b_left_buoy.position, right_b_pos=b_right_buoy.position)
+
+    defer.returnValue([ogrid, setup, target])
```

Comment:
Seems like you have two segments of nearly identical code for the two pair of buoys, consider making a function and calling it twice.